clients/src/main/java/org/apache/kafka/common/requests/AddPartitionsToTxnResponse.java

```diff
@@ -49,28 +52,37 @@ public class AddPartitionsToTxnResponse extends AbstractResponse {
     private final AddPartitionsToTxnResponseData data;
 
     private Map<TopicPartition, Errors> cachedErrorsMap = null;
+    
+    private Map<String, Map<TopicPartition, Errors>> cachedAllErrorsMap = null;
 
     public AddPartitionsToTxnResponse(AddPartitionsToTxnResponseData data) {
         super(ApiKeys.ADD_PARTITIONS_TO_TXN);
         this.data = data;
     }
 
+    // Only used for versions < 4
```

Comment:
I do wonder if we need both constructors. The old version is a subset of the new version where the number of producers happens to be 1.

Reply:
I left it this way for minimal code changes but we could get rid of this one.

Reply:
Yeah, the problem is that it makes the usage more confusing. I'd rather have a larger patch I guess and try to abstract some of the version details away.